hyper/contrib.py

```diff
@@ -43,28 +48,40 @@ def get_connection(self, host, port, scheme, cert=None):
         if cert is not None:
             ssl_context = init_context(cert=cert)
 
+        connection_key = (host, port, scheme, cert, proxy)
```

Comment:
Please let me know what do you think about https://github.com/Lukasa/hyper/pull/322/commits/bbf59ff8597f8be68c3fe43093762966e826050c

I don't like storing the whole dict as a key. It seems to bloat the key without much benefit